ledger/src/blockstore_processor.rs

```diff
@@ -142,237 +143,212 @@ fn get_first_error(
+                    Ok(ReplayResponse { result, timings }) => {
+                        results.push(result);
+                        new_timings.push(timings);
+                    }
+                    Err(_) => {
+                        error!("ReplayResponse recv error");
+                    }
+                },
+                Err(e) => {
+                    error!("error yooooo!!! error: {:?}", e);
+                }
+            }
+        }
 
-        let token_balances =
-            TransactionTokenBalancesSet::new(pre_token_balances, post_token_balances);
+        // timings.saturating_add_in_place(ExecuteTimingType::TotalBatchesLen, batches.len() as u64);
+        // timings.saturating_add_in_place(ExecuteTimingType::NumExecuteBatches, 1);
+        for timing in new_timings {
+            timings.accumulate(&timing);
+        }
 
-        transaction_status_sender.send_transaction_status_batch(
-            bank.clone(),
-            transactions,
-            execution_results,
-            balances,
-            token_balances,
-            rent_debits,
-        );
+        first_err(&results)?;
     }
 
-    let first_err = get_first_error(batch, fee_collection_results);
-    first_err.map(|(result, _)| result).unwrap_or(Ok(()))
+    Ok(())
 }
 
-fn execute_batches_internal(
-    bank: &Arc<Bank>,
-    batches: &[TransactionBatch],
-    entry_callback: Option<&ProcessCallback>,
-    transaction_status_sender: Option<&TransactionStatusSender>,
-    replay_vote_sender: Option<&ReplayVoteSender>,
-    timings: &mut ExecuteTimings,
-    cost_capacity_meter: Arc<RwLock<BlockCostCapacityMeter>>,
-) -> Result<()> {
-    inc_new_counter_debug!("bank-par_execute_entries-count", batches.len());
-    let (results, new_timings): (Vec<Result<()>>, Vec<ExecuteTimings>) =
-        PAR_THREAD_POOL.with(|thread_pool| {
-            thread_pool.borrow().install(|| {
-                batches
-                    .into_par_iter()
-                    .map(|batch| {
-                        let mut timings = ExecuteTimings::default();
-                        let result = execute_batch(
-                            batch,
-                            bank,
-                            transaction_status_sender,
-                            replay_vote_sender,
-                            &mut timings,
-                            cost_capacity_meter.clone(),
-                        );
-                        if let Some(entry_callback) = entry_callback {
-                            entry_callback(bank);
-                        }
-                        (result, timings)
-                    })
-                    .unzip()
-            })
+// for each index, builds a transaction dependency graph of indices that need to execute before
+// the current one.
+fn build_dependency_graphs(
+    tx_account_locks_results: &Vec<Result<TransactionAccountLocks>>,
+) -> Result<Vec<HashSet<usize>>> {
```

Comment:
If all the transactions conflict, then the resulting dependency graph would have all previous transactions in the set for each transaction.

Is this potentially a concern for performance, if a tx-scheduler queued up conflicting accounts into a single bundle/batch for example.

Reply:
yeah, it could result in worse performance + more memory usage. we'd prob be limited by account usage in the cost model